frontend/src/pages/PipelineSelector.tsx

```diff
@@ -113,6 +115,9 @@ class PipelineSelector extends React.Component<PipelineSelectorProps, PipelineSe
       logger.error('Could not get list of pipelines', errorMessage);
     }
 
+    // Warnings are showing up here in tests when the NewRun component is unmounted before being
+    // closed, saying setState is being called after the component is unmounted.
+    // The warning doesn't show up when actually using the app though.
```

Comment:
That function only exists, on `Page`, but it is very simple, so i can just emulate it here

Reply:
Yes, we should be looking at canceling promises on unmount, but that's an expensive fix. I don't see a lot of down sides to this, other than just be a little ugly.

Reply:
Yes, we probably should. I'll open an issue for properly fixing this, although we shouldn't really be holding our breath to do it. It might not be the best way to handle this, but it's not a bad fix IMO.

Reply:
Do you want me to do add that code to this component for this PR then?

Reply:
Up to you, it's probably not a lot of code so I'd do it, but it's fine to do this separately too.